### PR TITLE
Enable calling the indexers as functions returns promises

### PR DESCRIPTION
**Pull request by AyhamAlzoubi**
Use require instead of include

Use the native promise instead of bluebird

Added server for node apps will add the required routes

### CODE REVIEW

**Comment by AyhamAlzoubi** [comment]
@odino @davidino @unlucio can you please have a look ?

**Comment by AyhamAlzoubi** [comment]
This PR is to enable calling the indexer as a function

To avoid triggering it as a command every time we trigger it, to make it also flexible manageable from the server, I added the ability to call it as a function will return a promise, then we can know when it's done to do related things.

If you want to run it as a command it will also work fine

I added also a simple UI to trigger the indexers or to see the scheduled indexers running, you just need to pass the app (for nodejs projects only) to the server then it will add the required routes for you.

Please let me know if it looks fine from your side?

/cc @unlucio @davidino @odino

File: lib/zindex.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+var entitiesPath = config.get('dirs.indexers') || path.join(config.get('dirs.base') || config.get('rootDir'), 'indexers');
+var hl = require('highland');
+var lockFile = require('lockfile');
+var lockingFile = require('path').join(config.get('rootDir'), config.get('lockfile.name') || 'zindex.lock');
+
+function lock() {
+  console.log('Lock:: lockfile');
+  try {
+    lockFile.lockSync(locki);
```
Comment (hossam-fares) [inline lib/zindex.js]: No declaration for `locki`!

File: lib/zindex.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+  options = getBackendOptions(entity, options);
+
+  return utils.getFiles(backendsPath).then(function(backends) {
+    logger.debug('Indexer:: Found %d backends', backends.length);
+
+    logger.debug('Indexer:: indexIntoBackends:: promise resolved, going for transform... options: ' + JSON.stringify(data.options));
+    options = _.merge(options, data.options);
+
+    return transform(data.data, options).then(function(observable) {
```
Comment (hossam-fares) [inline lib/zindex.js]: `data.data` is bit misleading , let's make it `source.data`

**Pull request closed without merging**